src/lib/Microsoft.Health.Events/Model/EventMessage.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Events.Model
+{
+    public class EventMessage : IEventMessage
+    {
+        public EventMessage(string partitionId, DateTime dateTime)
+        {
+            EnqueuedTime = dateTime;
+            PartitionId = partitionId;
+        }
+
+        public EventMessage(
+            string partitionId,
+            ReadOnlyMemory<byte> body,
+            long sequenceNumber,
+            long offset,
+            DateTimeOffset enqueuedTime,
+            IDictionary<string, object> properties,
+            IReadOnlyDictionary<string, object> systemProperties)
+        {
```

Comment:
I'm wondering if it makes sense to do null checks here. It seems like some of these properties are "required", so enforcing this on the initializer might make sense. That or maybe in the factory.